aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -431,10 +431,10 @@ Unhealthy
 
 In some hosting scenarios, a pair of health checks are used that distinguish two app states:
 
-* The app is functioning but not yet ready to receive requests. This state is the app's *readiness*.
-* The app is functioning and responding to requests. This state is the app's *liveness*.
+* The app and its dependencies are functioning and able to process requests. This state is the app's *readiness*.
+* The app is functioning without checking dependencies. This state is the app's *liveness*.
 
-The readiness check usually performs a more extensive and time-consuming set of checks to determine if all of the app's subsystems and resources are available. A liveness check merely performs a quick check to determine if the app is available to process requests. After the app passes its readiness check, there's no need to burden the app further with the expensive set of readiness checks&mdash;further checks only require checking for liveness.
+The readiness check usually performs a more extensive and time-consuming set of checks to determine if all of the app's subsystems and resources are available. After the app passes its readiness check, there's no need to burden the app further with the expensive set of readiness checks. Ongoing checks only require checking for liveness.
```

Comment:
>  After the app passes its readiness check, there's no need to burden the app further with the expensive set of readiness checks. Ongoing checks only require checking for liveness.

This doesn't really make sense (I'm not sure how it got into the doc to begin with). 

External monitoring systems (for example kubernetes/docker) are the ones initiating the checks. It's not something the application author can control.

Reply:
Users are advised to make *all* health check implementations complete quickly and reliably.

Reply:
I'll research it and see. In the meantime, I'll remove that line and add your following remark.

Reply:
I think that I improperly based the remark on the following language from the engineering sample ...

> The readiness check might do a set of more expensive or time-consuming checks to determine if all other resources are responding. The liveness check performs a quick set of checks to determine if the process is functioning correctly.

I never used these types of checks before, and I assumed the wrong purpose. Unfortunately, I didn't see Damian's remarks at the time, and the text didn't get updated on review ...

> A failed liveness probe says: The application has crashed. You should shut it down and restart.
>
> A failed readiness probe says: The application is OK but not yet ready to serve traffic.

WRT ...

> how it got into the doc to begin with

It's been there since the topic was put up. You didn't remark on it at the time, so I'm not sure if you saw it or not ... you probably didn't see it or else you would've said something at the time.

Original PR (It's Line 376 of the *health-checks.md* file): https://github.com/dotnet/AspNetCore.Docs/pull/8959/files#diff-19855d53faf5f10a1834bfb535a7d831R376

I'll fix it up and get back to you :cat2::cat2::cat2: sometime this week.